Makefile

```diff
@@ -11,38 +11,68 @@ package_name=$(app_name)
 cert_dir=$(HOME)/.nextcloud/certificates
 version+=master
 
-all: appstore
+all: dev-setup build-js-production
+
+dev-setup: clean-dev npm-init
+
+dependabot: dev-setup npm-update build-js-production
 
 release: appstore create-tag
 
-create-tag:
-	git tag -a v$(version) -m "Tagging the $(version) release."
-	git push origin v$(version)
+build-js:
+	cd vue/ && npm run dev
 
-clean:
-	rm -rf $(build_dir)
-	rm -rf node_modules
+build-js-production:
+	cd vue/ && npm run build
+
+watch-js:
+	cd vue/ && npm run watch
+
+lint:
+	cd vue/ && npm run lint
+
+lint-fix:
+	cd vue/ && npm run lint:fix
 
-install-deps: install-npm-deps
+npm-init: npm-init-root npm-init-vue
 
-install-npm-deps:
-	npm install --production
+npm-init-root:
+	npm install
 
-install-npm-deps-dev:
-	npm install --deps
+npm-init-vue:
+	cd vue/ && npm install
 
-compile-handlebars-templates: dev-setup
+npm-update:
+	npm update
+	cd vue/ && npm update
+
+clean:
+	rm -f js/admin/*.js
+	rm -f js/admin/*.js.map
+	rm -f js/collections.js
+	rm -f js/collections.js.map
+	rm -f js/collectionsintegration.js
+	rm -f js/collectionsintegration.js.map
+	rm -rf $(build_dir)
+
+clean-dev: clean
+	rm -rf node_modules
+	cd vue/ && rm -rf node_modules
+
+compile-handlebars-templates:
 	bash compile-handlebars-templates.sh
 
-bundle-simplewebrtc: dev-setup
+bundle-simplewebrtc:
 	# webrtc-adapter uses JavaScript features not supported by browserify,
 	# so the sources need to be transformed using babel to a compatible
 	# version of JavaScript.
 	npx browserify --standalone SimpleWebRTC --transform [ babelify --global --presets [ @babel/env ] ] js/simplewebrtc/simplewebrtc.js > js/simplewebrtc/bundled.js
 
-dev-setup: install-npm-deps-dev
+create-tag:
+	git tag -a v$(version) -m "Tagging the $(version) release."
+	git push origin v$(version)
 
-appstore: clean install-deps
+appstore: dev-setup build-js-production compile-handlebars-templates bundle-simplewebrtc
```

Comment:
The compiled Handlebars templates and the bundled SimpleWebRTC are not removed when cleaning, so there is no need to call `compile-handlebars-templates` and `bundle-simplewebrtc`. Moreover, as those files are versioned the same version should end in the package, so I think that it would be better to remove those dependencies too to ensure that they are not mistakenly changed for the package.

Reply:
Yeah im not sure, the thing is, it would remind to check the stuff, as I see that before uploading and tagging, etc...